kotlinx-coroutines-core/build.gradle

```diff
@@ -184,13 +184,30 @@ jvmTest {
     minHeapSize = '1g'
     maxHeapSize = '1g'
     enableAssertions = true
-    systemProperty 'java.security.manager', 'kotlinx.coroutines.TestSecurityManager'
+    if (!Idea.active) {
+        // We should not set this security manager when `jvmTest`
+        // is invoked by IntelliJ IDEA since we need to pass
+        // system properties for Lincheck and stress tests.
+        // TODO this can be removed once IDEA is allowed to use `jvmLincheckTest` and
+        // TODO `jvmStressTest` tasks for the corresponding tests instead of using `jvmTest`.
+        systemProperty 'java.security.manager', 'kotlinx.coroutines.TestSecurityManager'
+    }
     // 'stress' is required to be able to run all subpackage tests like ":jvmTests --tests "*channels*" -Pstress=true"
     if (!Idea.active && rootProject.properties['stress'] == null) {
+        exclude '**/*LincheckTest.*'
         exclude '**/*StressTest.*'
     }
-    systemProperty 'kotlinx.coroutines.scheduler.keep.alive.sec', '100000' // any unpark problem hangs test
-
+    if (Idea.active) {
+        // Configure the IDEA runner for Lincheck and stress tests
+        // TODO this can be removed once IDEA is allowed to use `jvmLincheckTest` and
+        // TODO `jvmStressTest` tasks for the corresponding tests instead of using `jvmTest`.
+        jvmArgs = ['--add-opens', 'java.base/jdk.internal.misc=ALL-UNNAMED',   // required for transformation
+                   '--add-exports', 'java.base/jdk.internal.util=ALL-UNNAMED'] // in the model checking mode
+        maxHeapSize = '6g' // we need more memory for model checking
+        systemProperty 'kotlinx.coroutines.semaphore.segmentSize', '2' // better for testing in general
+        systemProperty 'kotlinx.coroutines.semaphore.maxSpinCycles', '1' // better for model checking
+        systemProperty 'kotlinx.coroutines.scheduler.keep.alive.sec', '100000' // any unpark problem hangs test
```

Comment:
Please, extract this configuration into a separate method. There's too much duplication.

Reply:
Fixed.